components/dropdown/index.jsx

```diff
@@ -0,0 +1,36 @@
+import { createElement } from "complate-stream";
+
+export default function Dropdown({ id, class: additionalClass, label, variant="secondary",
+									 active, disabled, type: typeName = "button", size }, ...children) {
+
+	let classNames = ["dropdown-toggle", "btn", "btn-" + variant];
+
+	if (additionalClass) {
+		classNames.push(additionalClass);
+	}
+
+	if (active) {
+		classNames.push("active");
+	}
+
+	if (disabled) {
+		classNames.push("disabled");
+	}
+
+	if (size) {
+		classNames.push(`btn-${size}`);
+	}
+
+	if (!typeName) {
+		typeName = "button";
+	}
+
+	return <div class="dropdown">
+		<button class={classNames.join(" ")} data-toggle="dropdown" type={typeName} id={id} aria-haspopup="true" aria-expanded="false">
+			{label || "Dropdown"}
```

Comment:
Do you think having defaults for user-facing text is a good idea?

Reply:
I can also see users wanting to use HTML here (for glyphs from icon fonts, say), so having `label` as an attribute might not be sufficiently powerful in the first place.

Reply:
OK, will remove default label. Maybe we will add icon-Attrbutes to some elements like buttons or dropdowns? Don't know how to support markup instead of plain text here.